eZ/Publish/Core/FieldType/Float/Type.php

```diff
@@ -222,7 +219,7 @@ protected function createValueFromInput($inputValue)
      */
     protected function checkValueStructure(BaseValue $value)
     {
-        if (!is_float($value->value)) {
+        if (!(is_float($value->value) || is_int($value->value))) {
```

Comment:
As we said integer should not be stored, it should not be allowed for `$value->value` to be of integer type.
Hence this change is not necassary.